pkg/registry/client.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/heroku/docker-registry-client/registry"
+	"github.com/okteto/okteto/pkg/log"
+)
+
+// New creates a new Registry with the given URL and credentials, then Ping()s it
+// before returning it to verify that the registry is available.
+func New(registryURL, username, password string) (*registry.Registry, error) {
+	transport := http.DefaultTransport
```

Comment:
can you add timeouts to this? I think the default transport doesn't have any.

Reply:
```
var DefaultTransport RoundTripper = &Transport{
	Proxy: ProxyFromEnvironment,
	DialContext: (&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
		DualStack: true,
	}).DialContext,
	ForceAttemptHTTP2:     true,
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * time.Second,
	TLSHandshakeTimeout:   10 * time.Second,
	ExpectContinueTimeout: 1 * time.Second,
}
```